### PR TITLE
Fix 3 discrepancies in ledger_entry

### LINKED ISSUE #738: Fix "ledger_entry" discrepancies
- "ledger_index" is an invalid index - should be `malformedRequest` instead of `invalidParams`
- "check" is an invalid index - should be `malformedRequest` instead of `invalidParams`
- "pay_channel", "deposit_preauth" is invalid account - should be `malformedRequest` instead of `malformedAddress`

### PR DESCRIPTION
**Pull request by godexsoft**
Fixes #738

### CODE REVIEW

File: src/rpc/handlers/LedgerEntry.h [hunk 1]
```diff
@@ -100,13 +100,17 @@ class LedgerEntryHandler
         static auto const rpcSpec = RpcSpec{
             {JS(binary), validation::Type<bool>{}},
             {JS(ledger_hash), validation::Uint256HexStringValidator},
-            {JS(ledger_index), validation::LedgerIndexValidator},
+            {JS(ledger_index),
```
Comment (cindyyan317) [inline src/rpc/handlers/LedgerEntry.h]: For "ledger_index", rippled returns "invalidParams". no need to change this one

File: src/rpc/handlers/LedgerEntry.h [hunk 2]
```diff
@@ -100,13 +100,17 @@ class LedgerEntryHandler
...
+             validation::WithCustomError{validation::LedgerIndexValidator, Status(ClioError::rpcMALFORMED_REQUEST)}},
             {JS(index), validation::Uint256HexStringValidator},
```
Comment (cindyyan317) [inline src/rpc/handlers/LedgerEntry.h]: index format error should return "malformedRequest"

File: src/rpc/handlers/LedgerEntry.h [hunk 3]
```diff
@@ -144,7 +148,9 @@ class LedgerEntryHandler
                      {JS(seq), validation::Required{}, validation::Type<uint32_t>{}},
                  },
              }},
-            {JS(payment_channel), validation::Uint256HexStringValidator},
+            {JS(payment_channel),
```
Comment (cindyyan317) [inline src/rpc/handlers/LedgerEntry.h]: I tested "ticket", "nft_page", "escrow" "offer" they all use malformedRequest for rippled.

File: src/rpc/handlers/LedgerEntry.h (hunk 1 above)
Comment (godexsoft) [reply src/rpc/handlers/LedgerEntry.h]: How did it make into the discrepancy list then 😆

**Review by cindyyan317 — APPROVED** [review]
LGTM

**Pull request merged**